plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,237 @@
+  Entity,
+} from '@backstage/catalog-model';
+import { Link, Progress } from '@backstage/core-components';
+
+/**
+ * Properties for an entity popover on hover of a component.
+ *
+ * @public
+ */
+export type EntityPeekAheadPopoverProps = PropsWithChildren<{
+  entityRef: CompoundEntityRef;
+}>;
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '30em',
+    },
+    descriptionTypography: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      display: '-webkit-box',
+      WebkitLineClamp: 2,
+      WebkitBoxOrient: 'vertical',
+    },
+  };
+});
+
+const maxTagChips = 4;
+
+const EmailCardAction = ({ email }: { email: string }) => {
+  return (
+    <Tooltip title={`Email ${email}`}>
+      <Button target="_blank" href={`mailto:${email}`} size="small">
+        <EmailIcon color="action" />
+      </Button>
+    </Tooltip>
+  );
+};
+
+const UserCardActions = ({ entity }: { entity: UserEntity }) => {
+  return (
+    <>
+      {entity.spec.profile?.email && (
+        <EmailCardAction email={entity.spec.profile.email} />
+      )}
+    </>
+  );
+};
+
+const GroupCardActions = ({ entity }: { entity: GroupEntity }) => {
+  return (
+    <>
+      {entity.spec.profile?.email && (
+        <EmailCardAction email={entity.spec.profile.email} />
+      )}
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+const EntityCardActions = ({ entity }: { entity: Entity }) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+
+  return (
+    <>
+      <Tooltip title="Show details">
+        <Link
+          component="button"
+          to={entityRoute({
+            name: entity.metadata.name,
+            namespace: entity.metadata.namespace || 'default',
+            kind: entity.kind.toLocaleLowerCase('en-US'),
+          })}
+        >
+          <InfoIcon color="action" />
+        </Link>
+      </Tooltip>
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+export const EntityPeekAheadPopover = ({
+  entityRef,
+  children,
+}: EntityPeekAheadPopoverProps) => {
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const popupState = usePopupState({
+    variant: 'popover',
+    popupId: 'entity-peek-ahead',
+  });
+
+  const [{ loading, error, value: entity }, load] = useAsyncFn(async () => {
+    const catalogApi = apiHolder.get(catalogApiRef);
+    if (catalogApi) {
+      const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+      if (!retrievedEntity) {
+        throw new Error(`${entityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [apiHolder, entityRef]);
+
+  useEffect(() => {
+    if (popupState.isOpen && !entity && !error && !loading) {
+      load();
+    }
+  }, [popupState.isOpen, load, entity, error, loading]);
+
+  return (
+    <>
+      <div data-testid="trigger" {...bindHover(popupState)}>
```

Comment:
Hm. I think this unconditionally being a `div` is problematic, or rather that it's `display: block`. Its only usage right now is to wrap a `Link` which is `display: inline`. The short term solution would be to just make this a `span` instead. Possibly we could make it possible to choose from the outside with an `inline: boolean` or `wrapper: React.ElementType` prop on `EntityPeekAheadPopover` or something like that, but for now it should at least probably be a `span`

Reply:
done.